airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_stream_reader.py

```diff
@@ -54,11 +54,22 @@ def filter_files_by_globs(files: List[RemoteFile], globs: List[str]) -> List[Rem
         """
         Utility method for filtering files based on globs.
         """
-        return [file for file in files if any(fnmatch(file.uri, glob) for glob in globs)]
+        if not globs:
+            return files
+
+        matches = {}
+
+        for file in files:
+            for g in globs:
+                if globmatch(file.uri, g, flags=GLOBSTAR):
```

Comment:
nit: can we add a comment here explaining that `GLOBSTAR` enables the recursive `**` matching? i had to look up what this did and if it saves an extra click for the next person might be worth it